.github/workflows/lint.yml

```diff
@@ -12,16 +12,16 @@ permissions:
 jobs:
   golangci:
     permissions:
-      contents: read  # for actions/checkout to fetch code
-      pull-requests: read  # for golangci/golangci-lint-action to fetch pull requests
+      contents: read # for actions/checkout to fetch code
+      pull-requests: read # for golangci/golangci-lint-action to fetch pull requests
     name: lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/setup-go@v3
+      - uses: actions/setup-go@v5
```

Comment:
Bit of a nitpick here, but will happily glance over it for this PR - could such changes be included as separate PRs? It makes them easier to review and diagnose when something goes wrong.

Reply:
> Bit of a nitpick here, but will happily glance over it for this PR - could such changes be included as separate PRs? It makes them easier to review and diagnose when something goes wrong.

opened #136 for separate PR.